eZ/Publish/Core/Repository/PermissionResolver.php

```diff
@@ -0,0 +1,266 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Repository;
```

Comment:
It does, I'll go for `eZ\Publish\Core\Repository\Permission` (singular), like in https://github.com/ezsystems/ezpublish-kernel/pull/1714 if you agree.